firebase-firestore/src/test/java/com/google/firebase/firestore/local/LocalStoreTestCase.java

```diff
@@ -1007,6 +1033,211 @@ public void testHandlesSetMutationThenAckThenTransformMutationThenAckThenTransfo
     assertChanged(doc("foo/bar", 2, map("sum", 3), Document.DocumentState.LOCAL_MUTATIONS));
   }
 
+  @Test
+  public void testUsesTargetMappingToExecuteQueries() {
+    // This test verifies that once a target mapping has been written, only documents that match
+    // the query are read from the RemoteDocumentCache.
+    assumeTrue(queryEngine instanceof IndexFreeQueryEngine);
```

Comment:
Nit: none of the other platforms have an equivalent to `assumeTrue`. Is this really better than just `if`/`return`?

It seems like `!garbageCollectorIsEager()` is a similar kind of assumption. Handle them the same way?

Reply:
I don't have a strong preference either but it seems like we should at least handle query engine and GC the same way. `assumeFalse(garbageCollectorIsEager)` or `assumeTrue(garbageCollectorNonEager())`?